docs/testing/troubleshooting.md

```diff
@@ -26,6 +26,25 @@ Logs do not affect the workbook.
 
 When your Excel Script encounters a problem running, it produces an error. You'll see a prompt pop-up asking if you want to **View Logs**. Press that button to open the console and display any errors.
 
+## Data limits
+
+There are hard limits set by the feature to prevent large amounts of data from being transferred at once and large numbers of individual transactions.
+
+### Excel
+
+Excel for the web has the following limitations when making calls to the workbook through a script:
+
+- Requests and responses are limited to **5MB**.
+- A range is limited to **five million cells**.
+
+If you're encountering errors when dealing with large datasets, try using multiple smaller ranges instead of larger ranges. You can also APIs like [Range.getSpecialCells](/javascript/api/office-scripts/excelscript/excelscript.range#getspecialcells-celltype--cellvaluetype-) to target specific cells instead of large ranges.
```

Comment:
So, batching essentially? Wondering if that bears calling out

Reply:
Hmm. That sort of terminology might not track with the script demographic. It's also weird because the async model has been referred to as the "batch" model and this is advocating the opposite: split the wire calls instead of combining them.